LibraryLoading.cpp

```diff
@@ -0,0 +1,50 @@
+#include "LibraryLoading.h"
+#include "Config.h"
+
+#include <string>
+#include <vector>
+
+#include <iostream>
+
+#include <llvm/Support/DynamicLibrary.h>
+#include <llvm/Support/FileSystem.h>
+
+#ifdef _WIN32
+#define DYLIB_EXTENSION ".dll"
+#elif __APPLE__
+#define DYLIB_EXTENSION ".dylib"
+#elif __linux__
+#define DYLIB_EXTENSION ".so"
+#else
```

Comment:
You can probably generate this with CMake if you're interested.

Reply:
Alternatively it'd be nice if there was something in LLVM that provided this or something similar.

Reply:
I couldn't find anything in LLVM that does this, so I'll configure it with CMake. I'll also make it just be blank if it's not any of those 3 just because usually OS's will append the extension on their own (we just do it because we also do manual searching of the filesystem).